lib/Common/Memory/HeapBucket.cpp

```diff
@@ -503,10 +504,42 @@ HeapBucketT<TBlockType>::SnailAlloc(Recycler * recycler, TBlockAllocatorType * a
     AllocationVerboseTrace(recycler->GetRecyclerFlagsTable(), _u("TryAlloc failed, forced collection on allocation [Collected: %d]\n"), collected);
     if (!collected)
     {
+        // wait for background sweeping finish if there are too many pages allocated during background sweeping
+        BOOL isBackgroundSweeping = recycler->IsConcurrentSweepExecutingState();
+        if (newAssignedPageCountDuringBGSweeping > (uint)CONFIG_FLAG(NewPagesCapDuringBGSweeping))
```

Comment:
I initially uses IsConcurrentExecutingState(), I remember it resulting worse perf in early-boyer (the code path will be hit more often). Let me try again since I made a few other adjustment to avoid the perf degradation.